src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/PackageModelFactory.cs

```diff
@@ -0,0 +1,195 @@
+// Updated PackageModelFactory class
+using System;
+using System.Collections.Generic;
+using NuGet.PackageManagement.VisualStudio;
+using NuGet.VisualStudio.Internal.Contracts;
+using ContractItemFilter = NuGet.VisualStudio.Internal.Contracts.ItemFilter;
+
+namespace NuGet.PackageManagement.UI.Models.Package
+{
+    internal class PackageModelFactory
+    {
+        private readonly INuGetSearchService _searchService;
+        private readonly INuGetPackageFileService _packageFileService;
+        private readonly IPackageVulnerabilityService _packageVulnerabilityService;
+        private readonly bool _includePrerelease;
+        private IReadOnlyCollection<PackageSourceContextInfo> _packageSources;
+
+        public PackageModelFactory(INuGetSearchService searchService, INuGetPackageFileService packageFileService, IPackageVulnerabilityService packageVulnerabilityService, bool includePrerelease, IReadOnlyCollection<PackageSourceContextInfo> packageSources)
+        {
+            _searchService = searchService ?? throw new ArgumentNullException(nameof(_searchService));
+            _packageFileService = packageFileService ?? throw new ArgumentNullException(nameof(_packageFileService));
+            _packageVulnerabilityService = packageVulnerabilityService ?? throw new ArgumentNullException(nameof(_packageVulnerabilityService));
+            _includePrerelease = includePrerelease;
+            _packageSources = packageSources ?? throw new ArgumentNullException(nameof(_packageSources));
+        }
+
+        public PackageModel Create(PackageSearchMetadataContextInfo metadata, ContractItemFilter itemFilter)
+        {
+            if (metadata == null)
+            {
+                throw new ArgumentNullException(nameof(metadata));
+            }
+
+            EmbeddedResourcesCapability embeddedResources = new EmbeddedResourcesCapability(_packageFileService, metadata.Identity!, metadata.ReadmeUrl);
+
+            if (metadata.PackagePath != null)
```

Comment:
Can we add the comment about PackagePath not being set for transitive packages here?
We should capture this is an issue as well.

Reply:
btw, this method looks so much cleaner now that we have methods such as CreateLocalPackageModel.

Reply:
Alternatively, can we just do metadata.TransitiveOrigins == null?

Reply:
> During the walkthrough we discussed init properties for the models, was the decision in the end to not use them?

For now, yes! 🙂 I started down the path of using init properties but the change was getting pretty big so we can still consider doing this later and I want to handle that as an iterative follow-up as this evolves.

Reply:
> Alternatively, can we just do metadata.TransitiveOrigins == null?

@nkolev92: I adjusted the logic. I prefer that over adding a comment and I think the new way looks much clearer to understand now. The tests still pass so the logic is doing the same as before. Let me know what you think! I saw your approval, so I won't block on this.

Reply:
This is better because there's no follow-ups either.